mongodbatlas/resource_mongodbatlas_api_key.go

```diff
@@ -92,8 +92,12 @@ func resourceMongoDBAtlasAPIKeyRead(ctx context.Context, d *schema.ResourceData,
 	orgID := ids["org_id"]
 	apiKeyID := ids["api_key_id"]
 
-	apiKey, _, err := conn.APIKeys.Get(ctx, orgID, apiKeyID)
+	apiKey, resp, err := conn.APIKeys.Get(ctx, orgID, apiKeyID)
 	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound || resp.StatusCode == http.StatusBadRequest {
+			d.SetId("")
+			return nil
+		}
```

Comment:
Could we add a warning saying that we could not find the key and we are silently removing the API key?

Reply:
If API key missing it will offer to create it during plan result so it would be removed from state after apply and replaced with new one could add message to catch that edge case

Reply:
Added a log message provider also shows this 
```
Note: Objects have changed outside of Terraform

Terraform detected the following changes made outside of Terraform since the last "terraform apply":

  # mongodbatlas_api_key.test has been deleted
```